packages/pg/test/integration/client/error-handling-tests.js

```diff
@@ -257,3 +257,18 @@ suite.test('cannot pass non-string values to query as text', (done) => {
     })
   })
 })
+
+if (!helper.args.native) {
+  suite.test('when a query has an invalid date binding', function (done) {
+    var client = createErorrClient()
+    var calledDone = false
+
+    client.query(new pg.Query({ text: 'SELECT $1::timestamp', values: [new Date(undefined)] }), function (err, res) {
+      if (!calledDone) {
+        calledDone = true
+        assert.equal(err.message, 'Query parameter value cannot be an invalid date.')
+        client.end(done)
+      }
+    })
```

Comment:
The `query` callback can only be called once (and if it weren’t, we wouldn’t want to silently ignore that):

```suggestion

    client.query(new pg.Query({ text: 'SELECT $1::timestamp', values: [new Date(undefined)] }), function (err, res) {
      assert.equal(err.message, 'Query parameter value cannot be an invalid date.')
      client.end(done)
    })
```

Reply:
It will cause this error, as the connection also gets terminated, and the callback will be executed again.

```
Message: 'Connection terminated' == 'Query parameter value cannot be an invalid date.'
AssertionError [ERR_ASSERTION]: 'Connection terminated' == 'Query parameter value cannot be an invalid date.
```

Reply:
That’s a serious bug, then.